### PR TITLE
[Enhancement] Reuse-schema-in-delta-writer

### LINKED ISSUE #7247: [Enhancement] Reuse-schema-in-delta-writer
## Enhancement
When an table convert a lot of columns, transfer `TabletSchema` to `Schema` through `convert_schema_to_format_v2` is very time consuming.
In fact under the same delta writer, `TabletSchema` are same, `Schema` are also same. So here we can reuse the `Schema` under the smae delta writer,
rather than repeatedly build the `Schema` many times in memtable.

Test Envrionment:

1. Table has 1008 rows,
2. Three concurrent stream load, each stram batch contains 2W rows, total 100 batch
3. 16 Concurrent flush thread

Before optimization: 943s
After optimization: 646s

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7247 

## Problem Summary(Required) ：

When an table convert a lot of columns, transfer `TabletSchema` to `Schema` through `convert_schema_to_format_v2` is very time consuming.
In fact under the same delta writer, `TabletSchema` are the same, and `Schema` is also the same. So here we can reuse the `Schema` under the same delta writer,
rather than repeatedly building the `Schema` many times in memtable.

Test Environment:

1. Table has 1008 columns,
2. Three concurrent stream load, each stram batch contains 2W rows, total of 100 batch
3. 16 Concurrent flush thread

Before optimization: 943s
After optimization: 646s

### CODE REVIEW

**Comment by meegoo** [comment]
> 1. Table has 1008 rows

1008 columns?

File: be/src/storage/memtable.h [hunk 1]
```diff
@@ -21,12 +21,11 @@ class MemTableSink;
 
 class MemTable {
 public:
-    MemTable(int64_t tablet_id, const TabletSchema* tablet_schema, const std::vector<SlotDescriptor*>* slot_descs,
-             MemTableSink* sink, MemTracker* mem_tracker);
-
-    MemTable(int64_t tablet_id, const Schema& schema, MemTableSink* sink, int64_t max_buffer_size,
+    MemTable(int64_t tablet_id, Schema* schema, const std::vector<SlotDescriptor*>* slot_descs, MemTableSink* sink,
```
Comment (meegoo) [inline be/src/storage/memtable.h:24]: MemTable UT should be update.

File: be/src/storage/memtable.h [hunk 2]
```diff
@@ -62,15 +63,13 @@ class MemTable {
...
-    vector<uint8_t> _result_deletes;
 
     // for sort by columns
     SmallPermutation _permutations;
     std::vector<uint32_t> _selective_values;
 
     int64_t _tablet_id;
-    Schema _vectorized_schema;
-    const TabletSchema* _tablet_schema;
+    Schema* _vectorized_schema;
```
Comment (chaoyli) [inline be/src/storage/memtable.h:72]:
```suggestion
const Schema* _vectorized_schema;
```

File: be/src/storage/memtable.cpp [hunk 3]
```diff
@@ -24,43 +24,48 @@ namespace starrocks::vectorized {
...
         : _tablet_id(tablet_id),
-          _tablet_schema(tablet_schema),
+          _vectorized_schema(schema),
           _slot_descs(slot_descs),
-          _keys_type(tablet_schema->keys_type()),
+          _keys_type(schema->keys_type()),
           _sink(sink),
           _aggregator(nullptr),
           _mem_tracker(mem_tracker) {
-    _vectorized_schema = std::move(ChunkHelper::convert_schema_to_format_v2(*tablet_schema));
```
Comment (chaoyli) [inline be/src/storage/memtable.cpp:45]: convert_schema_to_format_v2 is useless. We can remove the code also to eliminate the bad case.

Comment (xiaoyong-z) [reply be/src/storage/memtable.cpp:45]: lator i will submit a pr to remove all the convert_schema_to_format_v2

File: be/src/storage/delta_writer.cpp [hunk 4]
```diff
@@ -265,8 +266,12 @@ Status DeltaWriter::_flush_memtable() {
 }
 
 void DeltaWriter::_reset_mem_table() {
-    _mem_table = std::make_unique<MemTable>(_tablet->tablet_id(), _tablet_schema, _opt.slots, _mem_table_sink.get(),
-                                            _mem_tracker);
+    if (_schema_initialized == false) {
```
Comment (imay) [inline be/src/storage/delta_writer.cpp:269]:
```suggestion
    if (!_schema_initialized) {
```

**Comment by imay** [comment]
great job, LGTM

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**